### PR TITLE
Silence expected logging during `bundle-size` and  `test-status` tests

### PR DESCRIPTION
**Pull request by rcebulko**
Sets the standard log_level for tests to WARN
Drops log level to FATAL when testing operations that are expected to raise+log an error
Also fixes an `async` issue that kept Jest from exiting at the end of a test run

Drops the number of output lines for `bundle-size` tests from 436 to 4.
Drops the number of output lines for `test-status` tests from 66 to 4.

/cc @ampproject/wg-infra

### CODE REVIEW

File: bundle-size/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "deploy": "gcloud --project amp-bundle-size-bot app deploy",
     "setup-db": "node ./setup-db.js",
     "start": "probot run ./app.js",
-    "test": "jest"
+    "test": "LOG_LEVEL=warn jest"
```
Comment (danielrozenberg) [inline bundle-size/package.json]: That's not good, this means if I run `npm test` locally it will ignore the value from the environment. Remove this and set the LOG_LEVEL in Travis

File: bundle-size/test/api.test.js [hunk 2]
```diff
@@ -33,25 +33,29 @@ describe('bundle-size api', () => {
   let probot;
   let app;
   const db = dbConnect();
-  const nodeCache = new NodeCache();
+  let nodeCache;
+  let logWarnSpy;
 
-  beforeAll(async () => {
+  beforeAll(async done => {
```
Comment (danielrozenberg) [inline bundle-size/test/api.test.js]: Why was this change needed?

File: bundle-size/test/api.test.js [hunk 3]
```diff
@@ -33,25 +33,29 @@ describe('bundle-size api', () => {
...
     await setupDb(db);
 
     probot = new Probot({});
     app = probot.load(app => {
       const githubUtils = new GitHubUtils(new Octokit(), app.log, nodeCache);
       installApiRouter(app, db, githubUtils);
     });
+    // Stub app.log.warn to silence test log noise
+    logWarnSpy = jest.spyOn(app.log, 'warn').mockImplementation();
```
Comment (danielrozenberg) [inline bundle-size/test/api.test.js]: Why not just set LOG_LEVEL to be higher than `warn` in that case?

File: bundle-size/test/api.test.js [hunk 4]
```diff
@@ -33,25 +33,29 @@ describe('bundle-size api', () => {
...
     app.app = {
       getInstallationAccessToken: () => Promise.resolve('test'),
     };
+    done();
   });
 
-  beforeEach(async () => {
-    nodeCache.flushAll();
+  beforeEach(() => {
+    nodeCache = new NodeCache();
```
Comment (danielrozenberg) [inline bundle-size/test/api.test.js]: 🤔

File: bundle-size/test/api.test.js [hunk 5]
```diff
@@ -83,13 +87,17 @@ describe('bundle-size api', () => {
       .reply(200, getFixture('teams.listMembers.3065813'));
   });
 
-  afterEach(async () => {
+  afterEach(async done => {
```
Comment (danielrozenberg) [inline bundle-size/test/api.test.js]: Same, why do you need `done`?

File: bundle-size/test/api.test.js [hunk 6]
```diff
@@ -673,13 +681,19 @@ describe('bundle-size api', () => {
         )
         .reply(418, 'I am a tea pot');
 
+      // Silence expected error messages during testing.
+      const logLevel = probot.logger.level();
```
Comment (danielrozenberg) [inline bundle-size/test/api.test.js]: *Here* I would prefer that you mock it instead of changing the log level (so that if something changes in the code and the test fails, we can still get some idea as to why - i.e. by expecting the log to be have been called)

File: bundle-size/test/webhooks.test.js [hunk 7]
```diff
@@ -221,6 +221,11 @@ describe('bundle-size webhooks', () => {
         name: 'pull_request',
         payload: pullRequestPayload,
       });
+
+      // Silence expected error messages during testing.
+      const logLevel = probot.logger.level();
```
Comment (danielrozenberg) [inline bundle-size/test/webhooks.test.js]: ^

File: test-status/package.json [hunk 8]
```diff
@@ -17,7 +17,7 @@
     "deploy": "gcloud --project amp-test-status-bot app deploy",
     "setup-db": "node ./setup-db.js",
     "start": "probot run ./app.js",
-    "test": "jest"
+    "test": "LOG_LEVEL=warn jest"
```
Comment (danielrozenberg) [inline test-status/package.json]: ^

File: bundle-size/package.json (hunk 1 above)
Comment (rcebulko) [reply bundle-size/package.json]: Ahh good to know, done.

File: bundle-size/test/api.test.js (hunk 4 above)
Comment (rcebulko) [reply bundle-size/test/api.test.js]: `async` was not necessary; there's no `await`.
The `nodeCache` instance was leaving dangling `setTimeouts` the way it was being reused (Jest gave me warnings about this), and there's no need to share a cache between the tests

File: bundle-size/test/api.test.js (hunk 2 above)
Comment (rcebulko) [reply bundle-size/test/api.test.js]: Made these tweaks when I was trying to find the culprit preventing Jest from exiting properly. Reverting (the node cache was the issue)

File: bundle-size/test/api.test.js (hunk 5 above)
Comment (rcebulko) [reply bundle-size/test/api.test.js]: See above; reverted.

File: test-status/package.json (hunk 8 above)
Comment (rcebulko) [reply test-status/package.json]: Done

File: bundle-size/test/api.test.js (hunk 6 above)
Comment (rcebulko) [reply bundle-size/test/api.test.js]: Done.

File: bundle-size/test/webhooks.test.js (hunk 7 above)
Comment (rcebulko) [reply bundle-size/test/webhooks.test.js]: In this case, since you want the error to propagate outside the Probot handler, the message is automatically logged by Probot's error event handler (as opposed to the app/context/request loggers). As best as I can tell, there's no direct access to these methods (they're either private or non-explicit, somehow embedded in Probot error catching code). The change in log level still takes effect, though.

File: bundle-size/test/api.test.js (hunk 3 above)
Comment (rcebulko) [reply bundle-size/test/api.test.js]: I opted for this because every single test logs a warning about the bundle size not existing, but if you had logging enabled for debugging purposes (at INFO level), you'd still be able to see all the info logs.

Comment (danielrozenberg) [reply bundle-size/test/api.test.js]: Ack

File: bundle-size/test/webhooks.test.js (hunk 7 above)
Comment (danielrozenberg) [reply bundle-size/test/webhooks.test.js]: Ack
